src/api-service/__app__/onefuzzlib/secrets.py

```diff
@@ -79,4 +79,4 @@ def get_secret_obj(secret_url: str, model: Type[A]) -> A:
 def delete_secret(secret_url: str) -> None:
     (vault_url, secret_name) = parse_secret_url(secret_url)
     keyvault_client = get_keyvault_client(vault_url)
-    keyvault_client.begin_delete_secret(secret_name).wait()
```

Comment:
Doesn't this change the semantics of this function? Are we just removing the `wait()` for the sake of the bad type signature?

Reply:
If we really don't care that we blocked until deletion completed, I'd suggest renaming this method and documenting that it only starts deletion, and does not allow callers to determine when deletion has actually completed.

The main thing that isn't clear to me is if the deletion operation can ever fail after the Azure SDK has given us an `LROPoller`.

Reply:
This does not change the semantics of the function.  Previously, this was a blocking operation without ever checking the result.   

 `result()` needs to be called to get the result.

Reply:
Understood re `result()`, but I guess my point is that before, after calling this, our deletion request either succeeded or failed (though we couldn't tell which). Now neither may have happened. I agree that from a caller perspective, this difference probably isn't actionable, due to the lack of any result value or exception.

I still think it would be prudent to rename this to emphasize that this just _attempts_ an operation. I'm happy that the uncertainty is at least more obvious in the implementation now, though.